middle_end/flambda2/simplify/inlining/call_site_inlining_decision.ml

```diff
@@ -142,14 +142,22 @@ let argument_types_useful dacc apply =
       (Apply.args apply)
 
 let might_inline dacc ~apply ~code_or_metadata ~function_type ~simplify_expr
-    ~return_arity : Call_site_inlining_decision_type.t =
+    ~return_arity ~rec_depth_exceeded : Call_site_inlining_decision_type.t =
   let denv = DA.denv dacc in
   let env_prohibits_inlining = not (DE.can_inline denv) in
   let decision =
     Code_or_metadata.code_metadata code_or_metadata
     |> Code_metadata.inlining_decision
   in
-  if Function_decl_inlining_decision_type.must_be_inlined decision
+  if Function_decl_inlining_decision_type.is_stub decision
```

Comment:
Since we already have the code metadata in hand, is there a reason to say `is_stub decision` here rather than just check whether the function is a stub? It seems much clearer for the policy to be “inline stubs even if it would exceed the max recursion depth” than “inline those functions for which the definition-site inlining decision was true for reasons of being a stub, even if it would exceed the max recursion depth.”